src/util/anim_ease.ts

```diff
@@ -1,399 +1,323 @@
-    return c * Math.sqrt(1 - (t = t / d - 1) * t) + b;
+  if (!p) {
+    p = d * 0.3;
   }
-
-  /**
-   * Circular easing in and out
-   * @memberOf fabric.util.ease
-   */
-  function easeInOutCirc(t, b, c, d) {
-    t /= d / 2;
-    if (t < 1) {
-      return -c / 2 * (Math.sqrt(1 - t * t) - 1) + b;
-    }
-    return c / 2 * (Math.sqrt(1 - (t -= 2) * t) + 1) + b;
+  const opts = normalize(a, c, p, s);
+  return opts.a * 2 ** (-10 * t) * Math.sin((t * d - opts.s) * (twoMathPi) / opts.p ) + opts.c + b;
+}
+
+/**
+ * Elastic easing in and out
+ * @memberOf fabric.util.ease
+ */
+export const easeInOutElastic = (t, b, c, d) => {
+  const s = 1.70158, a = c;
+  let p = 0;
+  if (t === 0) {
+    return b;
   }
-
-  /**
-   * Elastic easing in
-   * @memberOf fabric.util.ease
-   */
-  function easeInElastic(t, b, c, d) {
-    var s = 1.70158, p = 0, a = c;
-    if (t === 0) {
-      return b;
-    }
-    t /= d;
-    if (t === 1) {
-      return b + c;
-    }
-    if (!p) {
-      p = d * 0.3;
-    }
-    var opts = normalize(a, c, p, s);
-    return -elastic(opts, t, d) + b;
+  t /= d / 2;
+  if (t === 2) {
+    return b + c;
   }
-
-  /**
-   * Elastic easing out
-   * @memberOf fabric.util.ease
-   */
-  function easeOutElastic(t, b, c, d) {
-    var s = 1.70158, p = 0, a = c;
-    if (t === 0) {
-      return b;
-    }
-    t /= d;
-    if (t === 1) {
-      return b + c;
-    }
-    if (!p) {
-      p = d * 0.3;
-    }
-    var opts = normalize(a, c, p, s);
-    return opts.a * Math.pow(2, -10 * t) * Math.sin((t * d - opts.s) * (2 * Math.PI) / opts.p ) + opts.c + b;
+  if (!p) {
+    p = d * (0.3 * 1.5);
   }
-
-  /**
-   * Elastic easing in and out
-   * @memberOf fabric.util.ease
-   */
-  function easeInOutElastic(t, b, c, d) {
-    var s = 1.70158, p = 0, a = c;
-    if (t === 0) {
-      return b;
-    }
-    t /= d / 2;
-    if (t === 2) {
-      return b + c;
-    }
-    if (!p) {
-      p = d * (0.3 * 1.5);
-    }
-    var opts = normalize(a, c, p, s);
-    if (t < 1) {
-      return -0.5 * elastic(opts, t, d) + b;
-    }
-    return opts.a * Math.pow(2, -10 * (t -= 1)) *
-      Math.sin((t * d - opts.s) * (2 * Math.PI) / opts.p ) * 0.5 + opts.c + b;
+  const opts = normalize(a, c, p, s);
+  if (t < 1) {
+    return -0.5 * elastic(opts, t, d) + b;
   }
-
-  /**
-   * Backwards easing in
-   * @memberOf fabric.util.ease
-   */
-  function easeInBack(t, b, c, d, s) {
-    if (s === undefined) {
-      s = 1.70158;
-    }
-    return c * (t /= d) * t * ((s + 1) * t - s) + b;
+  return opts.a * Math.pow(2, -10 * (t -= 1)) *
+    Math.sin((t * d - opts.s) * (twoMathPi) / opts.p ) * 0.5 + opts.c + b;
+}
+
+/**
+ * Backwards easing in
+ * @memberOf fabric.util.ease
+ */
+export const easeInBack = (t, b, c, d, s = 1.70158) => c * (t /= d) * t * ((s + 1) * t - s) + b;
+
+/**
+ * Backwards easing out
+ * @memberOf fabric.util.ease
+ */
+export const easeOutBack = (t, b, c, d, s = 1.70158) => c * ((t = t / d - 1) * t * ((s + 1) * t + s) + 1) + b;
```

Comment:
I see this number a lot. Maybe define it as const at top of file?

Reply:
yes there are a bunch of numbers. i wanted to define them, then i realized i didn't know how to call them and moved on for now

Reply:
I totally get that